cmd/saga/main.go

```diff
@@ -2,14 +2,161 @@ package main
 
 import (
 	"fmt"
+	saga "github.com/einride/saga/cmd/saga/gen"
+	"github.com/einride/saga/cmd/saga/internal/auth"
+	"github.com/spf13/cobra"
+	"go.einride.tech/aip-cli/aipcli"
 	"os"
+)
 
-	saga "github.com/einride/saga/cmd/saga/gen"
+const (
+	authFlagAnnotation = "saga_annotation_auth_flag"
+	addressFlag        = "address"
+	tokenFlag          = "token"
+	clientIDFlag       = "client-id"
+	clientSecretFlag   = "client-secret"
+
+	setUserCmdName   = "set-user"
+	clearUserCmdName = "clear-user"
 )
 
 func main() {
-	if err := saga.NewModuleCommand("saga", "Saga CLI").Execute(); err != nil {
-		_, _ = fmt.Fprintln(os.Stderr, err)
-		os.Exit(1)
+	moduleCmd := saga.NewModuleCommand(
+		"saga",
+		"Saga CLI",
+		setUserCommand(),
+		clearUserCommand(),
+	)
+
+	inputArgs := os.Args[1:]
+	invokedCmd, _, err := moduleCmd.Find(inputArgs)
+	if err != nil {
+		exitOnError(err)
+	}
+
+	// PersistentPreRun is used in aip-cli-go
+	originalPreRun := invokedCmd.PersistentPreRun
+	// PersistentPreRunE is picked over PersistentPreRun if both defined
+	invokedCmd.PersistentPreRunE = func(cmd *cobra.Command, args []string) error {
+		if invokedCmd.Name() == setUserCmdName || invokedCmd.Name() == clearUserCmdName {
+			return nil
+		}
+		if originalPreRun != nil {
+			originalPreRun(cmd, args)
+		}
```

Comment:
Using PersistentPreRunE over PersistentPreRun seems to be the current behavior in cobra but not a guarantee in their API from what I can see 😅 

Perhaps we should instead clear out the original one so that we're safe from changes in cobra.

Also aip-cli-go could change, so perhaps clear out any such function as well 😬 

```suggestion
	// Keep any PersistentPreRun functions set by aip-cli-go
	originalPreRun := invokedCmd.PersistentPreRun
	invokeCmd.PersistentPreRun = nil
	originalPreRunE := invokeCmd.PersistentPreRunE
	invokedCmd.PersistentPreRunE = func(cmd *cobra.Command, args []string) error {
		if invokedCmd.Name() == setUserCmdName || invokedCmd.Name() == clearUserCmdName {
			return nil
		}
		if originalPreRunE != nil {
			if err := originalPreRunE(cmd, args); err != nil {
				return err
			}
		} else if originalPreRun != nil {
			originalPreRun(cmd, args)
		}
```

Reply:
Thanks, I like that much better